src/hooks/useContributions.ts

```diff
@@ -38,7 +38,98 @@ const mockData: UserContributions = {
   ],
 };
 
-export function useContributionUsers(): UserContributions {
+export function useContributionUsers(timeInput: {
+  start_date: string;
+  end_date: string;
+}): UserContributions {
+  const integrations = useCurrentCircleIntegrations();
+  const responses = useQueries(
+    integrations.data
+      ? integrations.data
+          .filter(
+            integration =>
+              integration.type === 'dework' || integration.type === 'wonder'
+          )
+          .map(integration => ({
+            queryKey: `circle-integration-contributions-${integration.id}-${timeInput.start_date}-${timeInput.end_date}`,
+            queryFn: () => {
+              if (integration.type === 'dework') {
+                return fetch(
+                  `https://api.deworkxyz.com/integrations/coordinape/${
+                    integration.data.organizationId
+                  }?epoch_start=${timeInput.start_date}&epoch_end=${
+                    timeInput.end_date
+                  }&workspace_ids=${encodeURIComponent(
+                    integration.data.workspaceIds?.join(',') || ''
+                  )}`
+                )
+                  .then(res => res.json())
+                  .then(res => res as Response);
+              }
+              if (integration.type === 'wonder') {
+                let url = `https://external-api.wonderapp.co/v1/coordinape/contributions?org_id=${integration.data.organizationId}&epoch_start=${timeInput.start_date}&epoch_end=${timeInput.end_date}`;
+                if (integration.data.podIds) {
+                  for (const podId of integration.data.podIds) {
+                    url += `&pod_ids=${podId}`;
+                  }
+                }
+                return fetch(url)
+                  .then(res => res.json())
+                  .then(res => res as Response);
+              }
+            },
+          }))
+      : []
+  );
+  /**
+   * responses are individual responses from each integration
+   * looping over the responses from various integration stiching together to make a userContributions object that's easier to work eith
+   */
+
+  return useMemo(() => {
+    const combinedContribution: UserContributions = {};
```

Comment:
is there any way to sort these?

Reply:
why sort them? That sounds unnecessarily expensive. Indexing them makes more sense, no?